Jenkinsfile

```diff
@@ -1,29 +1,30 @@
 elifePipeline {
     def commit
+    stage 'Checkout', {
+        checkout scm
+        commit = elifeGitRevision()
+    }
+
     elifeOnNode(
         {
-            stage 'Checkout', {
+            stage 'Container image', {
                 checkout scm
-                commit = elifeGitRevision()
+                sh 'docker-compose -f docker-compose.ci.yml build'
             }
 
-            stage 'Container image', {
-                sh 'docker-compose -f docker-compose.ci.yml build'
-                sh 'chmod 777 build/ && docker-compose -f docker-compose.ci.yml run ci ./project_tests.sh'
-                step([$class: "JUnitResultArchiver", testResults: 'build/phpunit.xml'])
-                sh 'docker-compose -f docker-compose.ci.yml run ci ./smoke_tests.sh web'
+            stage 'Project tests', {
+                try {
+                    sh 'chmod 777 build/ && docker-compose -f docker-compose.ci.yml run --rm ci ./project_tests.sh'
+                    step([$class: "JUnitResultArchiver", testResults: 'build/phpunit.xml'])
+                    sh 'docker-compose -f docker-compose.ci.yml run --rm ci ./smoke_tests.sh web'
+                } finally {
+                    sh 'docker-compose -f docker-compose.ci.yml stop'
```

Comment:
Otherwise the containers stay up, occupying CPU, memory, ports and preventing further builds from running.